src/main/java/com/shade/decima/ui/data/viewer/texture/TextureViewer.java

```diff
@@ -45,19 +57,86 @@ public JComponent createComponent() {
 
     @Override
     public void refresh(@NotNull JComponent component, @NotNull Editor editor) {
-        final RTTIObject value = (RTTIObject) Objects.requireNonNull(((CoreEditor) editor).getSelectedValue());
+        Object obj = Objects.requireNonNull(((CoreEditor) editor).getSelectedValue());
         final PackfileManager manager = ((CoreEditor) editor).getInput().getProject().getPackfileManager();
-
-        final HwTextureHeader header = value.<RTTIObject>get("Header").cast();
-
-        final TextureViewerPanel panel = (TextureViewerPanel) component;
-        panel.setStatusText("%sx%s (%s, %s)".formatted(header.getWidth(), header.getHeight(), header.getType(), header.getPixelFormat()));
-
-        SwingUtilities.invokeLater(() -> {
-            panel.getImagePanel().setProvider(getImageProvider(value, manager));
-            panel.getImagePanel().fit();
-            panel.revalidate();
-        });
+        final Project project = ((CoreEditor) editor).getInput().getProject();
+        final EnumSet<Channel> channels = EnumSet.noneOf(Channel.class);
+        CoreBinary binary = ((CoreEditor) editor).getBinary();
+        String textureUsageName = "";
+        String colorSpace = "";
+
+        try {
+            RTTIObject value = (obj instanceof RTTIReference ref) ? ref.get(project, binary) : (RTTIObject) obj;
+
+            if (value.type().getTypeName().equals("TextureBindingWithHandle")) {
+                final RTTITypeEnum textureSetTypeEnum = project.getTypeRegistry().find("ETextureSetType");
+                final RTTIReference textureResourceRef = value.ref("TextureResource");
+                final RTTIReference.FollowResult textureResource = textureResourceRef.follow(project, binary);
+                final int usageType = value.i32("PackedData") >>> 2 & 0xf;
+                textureUsageName = textureSetTypeEnum.valueOf(usageType).name();
```

Comment:
I had copied that from the texture export part in the dmf exporter. Later I made PackingInfoHandler#getPurpose public and forget to update that part. Was there a reason why  textureSetTypeEnum was not used in the PackingInfoHandler#getPurpose function and instead the mapping from value to string is done via switch case? Would it not be better to rely on the Enum mapping as it is define in the json type file and can be potential different between the different games. The same maybe also valid for PackingInfoHandler#getChannel. I will change this to use getPurpose. How have you found that PackedData contains texture purpose and need to be shifted by two bytes. Do you have an idea, for what the rest bits are used. In HZD they are mostly 0xf or 0x7 other I have so far not seen. I have also found some texture binding entries, where there is no matching usageName found in the packingInfo of the referenced TextureSet. e.g see aloy_ct_a_lx.core (Group by Type) -> RenderEffectResource[2] -> TextureBindings[23]: PackedData show usage=Translucency_Amount but aloy_set.core TextureSet has only an entry with usage=Translucency_Diffusion. And what also is strange that the TextureBindings does not have any entries referencing the Roughness, Reflectance, AO of the TextureSet. So the texture handling of the game is still a mystery. In the HZDMeshTool plugin for Blender I relied so far only on the packingInfo in the TextureSet, as I didn't not know about the packedData at that time to generate the needed texture nodes for the material.

Reply:
The `getPurpose` method exists because it's called in a context without access to the `Project`/`RTTIRegistry`.

Unfortunately, I'm unaware of the meaning of the `PackedData` field bits.